### PR TITLE
Issue #1363: Adding option for exlusion time of failed requests from time metrics

### PR DESCRIPTION
**Pull request by vs-odessa**
This PR revives the request from https://github.com/locustio/locust/issues/1363 (_If possible, please reopen it_)

It allows adding the parameter  `--exсlude-failures` to locust runner to ignore response times of failed requests from time-based metrics (min,max, avg, percentiles etc.)

This helps to exclude abnormal values (too big or too small) from time statistics produced by failed requests. (For example, when getting gateway timeout errors equal to 60 seconds or more)
At the same time, it keeps stats for the total requests number and failures number.

**Example:** 
9 passed requests with 1 second response time + 1 failed request with 5 seconds response time

Without option:
`Avg time = (9*1 + 1*5) / 10 = 1.4 sec, Max time = 5 sec`

With option `--exclude-failures`:
`Avg time = (9*1) / 9 = 1 sec, Max time = 1 sec`

Total/Failed = 10/1 in both cases

**Update:** In a second commit added a feature for setting or overriding global setting for exclusion failures by passing keyword parameter to specified request(s).

### CODE REVIEW

**Comment by vs-odessa** [comment]
@heyman, @cyberw - please review

**Comment by vs-odessa** [comment]
I'm not sure that it can be solved by adding a custom handler - we can add our own actions, but it will not discard a core handler that writes all requests to stats. Correct me if I am wrong, but looks like a custom class should be created from Runner with an overriding on_request method, and this will require a significant refactoring of existing projects. 

Let me add a few words about the motivation for this feature with some examples.

In my experience - I have at least a third project where this feature would be extremely useful, and I saw similar demands from other engineers.
It is true, that most of the tools have such behaviour as Locust by default, but I guess all of them came from times when synchronous HTTP requests were >99% of load tools' purpose.
In modern event-based systems, there is very often a situation when you need to check asynchronous transactions.
In this case, we are interested not only in a single request time, but need also to check a status of a background service task produced by the initial request - until the service task is finished.
Usually, such flow is wrapped as "locust request" to get its duration and other metrics. This flow includes retries of some requests with a timeout, and if something went wrong we get failed transaction with a time equal to timeout.
This significantly exceeds the time for successful operations and distorts a picture of the positive system performance.
For the same reason, you can't rely on minimum/maximum value if any failures occurred in requests.
And conversely, various client & server errors often may give unusually fast timings. In this case, we get a curious situation: the greater number of failures - the better performance metrics appear.

Summing up, I believe it's nice to have this option from the box, and it will be often used. It shouldn't break any old functionality and adds a very slight overhead in the handling of failed requests only.

**Comment by cyberw** [comment]
I get that you have a use case. Can it be worked around? If not then the PR may be justified, otherwise I kinda disagree.

**Comment by vs-odessa** [comment]
I've analyzed and played for a while with a locust code trying to implement a workaround, and in my view, it looks unlikely feasible :(
Environment class and Runner class, that manages requests stats - both are core features and they are not managed when running Locust in a standalone mode.
In theory, perhaps there's some workaround way for "running as a library" mode, but it also looks too complicated and hard maintainable.
If I overlooked something and you see an idea for a solution, please point me in the right direction for it.

**Comment by vs-odessa** [comment]
Possible workaround using response validation:
`        with self.client.request(method, url, name=name, catch_response=False) as response:
            if 'should-fail' in url:
                response.request_meta['response_time'] = None`
